### PR TITLE
[New Operator][PyTorch] AdaptiveAvgPool

### PR DESCRIPTION
**Pull request by jackm321**
Summary:
Add the PyTorch AdaptiveAvgPool op and add loading from PyTorch [adaptiveavgpool2d](https://pytorch.org/docs/stable/nn.html#adaptiveavgpool2d) op.

Documentation:
doxygen

Test Plan:
Added unit tests
`python setup.py test`

### CODE REVIEW

File: lib/Exporter/ONNXModelWriter.cpp [hunk 1]
```diff
@@ -691,6 +691,18 @@ ONNXModelWriter::writeSparseToDenseMask(const SparseToDenseMaskNode *node,
   return writeAllWithNode(node->getName(), node, proto);
 }
 
+llvm::Error
+ONNXModelWriter::writeAdaptiveAvgPool(const AdaptiveAvgPoolNode *node,
```
Comment (yinghai) [inline lib/Exporter/ONNXModelWriter.cpp]: We still need this?

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 2]
```diff
@@ -445,6 +445,42 @@ void PyTorchModelLoader::loadMaxPool2d(const torch::jit::Node *ptNode) {
...
+
+  // Indexes of aten::adaptive_avg_pool2d inputs.
+  struct Inputs {
+    enum {
+      input = 0, // NCHW
+      output_size = 1,
+    };
+  };
+
+  // Glow expects inputs to be in NHWC but PyTorch keeps them in NCHW so we
```
Comment (yinghai) [inline torch_glow/src/PyTorchModelLoader.cpp]: Glow works all in NHWC?

File: lib/Exporter/ONNXModelWriter.cpp (hunk 1 above)
Comment (jackm321) [reply lib/Exporter/ONNXModelWriter.cpp]: Yeah actually this just landed yesterday #3167

File: torch_glow/src/PyTorchModelLoader.cpp (hunk 2 above)
Comment (jackm321) [reply torch_glow/src/PyTorchModelLoader.cpp]: Yes

**Pull request closed without merging**